crates/matrix-sdk/src/room/mod.rs

```diff
@@ -3141,6 +3147,52 @@ impl Room {
         Ok(())
     }
 
+    /// Subscribe to live location sharing events for this room.
+    ///
+    /// The returned receiver will receive a new event for each sync response
+    /// that contains a 'm.beacon' event.
+    pub fn subscribe_to_live_location_shares(
+        &self,
+    ) -> (JoinHandle<()>, broadcast::Receiver<LiveLocationShare>) {
+        let (sender, receiver) = broadcast::channel(128);
+
+        let client = self.client.clone();
+        let room_id = self.room_id().to_owned();
+        let room = self.clone();
+
+        let handle: JoinHandle<()> = spawn(async move {
+            let beacon_event_handler_handle = client.add_room_event_handler(&room_id, {
```

Comment:
Wait, you should not spawn a new task. `Client::add_room_event_handler` returns an `EventHandlerHandle`. You should rely on this type. Spawning a task will just run `add_room_event_handler` and that's it, nothing more. I don't see the point of doing that.

Your function should return a `(EventHandlerHandle, Receiver)`, or a new type wrapping that. The new type can remove the event handler on drop for example, it seems a bit cleaner. Otherwise, the event handler is never removed from the client.

Reply:
I will work on integrating this before I squash my commits. Thanks for the detailed response!

Reply:
Updated commit with your suggested changes. Let me know if it aligns properly with your idea of new type wrapping `(EventHandlerHandle, Receiver)`